### ISSUE TITLE
BT emails delivery issue

### ISSUE THREAD

**Issue by JimKillock** [openrightsgroup/Blocking-Middleware#116]
BT Categorisation report that:

"We have recently noticed that when we reply back to emails from blocked.org.uk we intermittently received undeliverable messages from the mail servers. I have attached a few mails whereby we have replied back and received either a Delivery Delayed or Undeliverable."

Can you check to see what has happened?

**Comment by JimKillock** [comment]
You will have seen the emails and chat: can you change the way we forward emails so that the email:

1. Uses a From: header that belongs to @blocked.org.uk 
2. Allows the user to reply to the same From: header, and route that back to categorisation@bt.com ; 
3. The BT reply to should consistent with the original review request so any threads can be preserved.

**Comment by dantheta** [comment]
We can do (1) and (2) pretty easily, we'd just need to set up a fixed `reply-` alias for each of the ISPs.  Capturing subsequent ISP replies (3) is slightly harder, as we won't have the original ISP report alias available after their reply in (2), since that reply will have used the user's own email address.  Looking up the correct alias based on the subject line and ISP should be possible.

**Comment by JimKillock** [comment]
I think I see what you mean! 

Nevertheless with (3) the user would reply with their own email address, rather than user-123@blocked.org.uk but the reply-to, eg BT-123@blocked.org.uk would be the same as the email sent "as if" BT, so should match that way? You'd need a unique "User Alias" and a unique "BT Alias" on our end?

**Comment by JimKillock** [comment]
(I've edited the comment and the task to make this a bit clearer. I think my descrption wasn't quite clear.)

**Comment by JimKillock** [comment]
Does this make sense @dantheta ? We should be able to play 'man in the middle' and rewrite the "from" in both directions, in essence.

**Comment by dantheta** [comment]
Yep - I'm working on the required postfix configuration at the moment.

**Comment by dantheta** [comment]
The site now adds a reply-isp-xxxxxx `From:` address to email coming from the ISPs.  BT still include a reply-to header in their outgoing mail, so we might not be able to capture all of the user -> ISP replies (though there haven't been many of those).

It's been tested reasonably well, but if it causes any problems we can revert the change quite easily.

**Comment by JimKillock** [comment]
Sounds good Daniel; can we consider changing the reply-to header as well?
